### ISSUE TITLE
Determine if already signed in

### ISSUE THREAD

**Issue by RFannin** [IdentityServer/IdentityServer3#1821]
Is it possible to determine if a user has signed in via another client? For example, the user goes to client A and logs in. Is it possible to determine on client B that the user has an active SSO session when they go to client B?

**Comment by pawepaw** [comment]
If user will try to log in into client B he will be automatically logged without need to provide credentials, because he is already signed into IdentityServer and cookie exist. Or maybe you are asking if you can do it implicitly when going to client B?

**Comment by RFannin** [comment]
I was wanting to initiate the logon for them if they had previously signed in to a different client. Otherwise, the user would have to initiate the log in. Is that a silly use case?

**Comment by farshidpourlatifi** [comment]
This is my question too.

Here is the situation: (Client refers to identityserver client, which is a web app on some domain)
Client A and B are on different domains.
Client A and B, both have two faces: 
1. Anonymous
2. Authorized

When users log-in, in client A they see the the Authorized face.
Now if they browse client B, they see the Anonymous face.
If we send them to an unauthorized resource (page eg.) they go through the identityserver sso and get back to B and they are signed in, but this is not the case.

What I need (as RFannin dose I think) is a way to automatically log-in user on first hit in client B so they view the Authorized face there too.

Another use-case would be a JavaScript widget hosted on client B, which will be loaded in client A, while the user is logged-in in client A, they see Anonymous face of the widget hosted on client B.

I know that there are JS clients to use identityserver, but what I need is not authorizing the user client, but the application on Server B.

I do not know what this is called! or if this is a common problem at all.
SSO with automatic sign in? maybe...

Any way I am seeking the answer for now, if any one can help on this or give us some tips, or shed some light on the possible solutions, it would be appreciated.

**Comment by farshidpourlatifi** [comment]
What we have in mind to achieve this is here:

Host a JavaScript file in identityserver3 which will output the user currently logged in in identityserver, the out put will be somesing like this:

```javascript
var user = {
      isAuthenticated : true,
      email : 'user@domain.com'
};
```

Then have this JavaScript loaded in all clients.
just after that script, have another one from the client domain which will check it's own user status with the one from identityserver and take proper action.

So please let me know if this is possible, or if anyone has already done something similar.

**Comment by brockallen** [comment]
> If we send them to an unauthorized resource (page eg.) they go through the identityserver sso and get back to B and they are signed in, but this is not the case.

If this is not working for you, then you need sort this part out first. 

As for the larger design, if you need to know if the user is logged in or not, you can make a request to IdSvr and pass the prompt=none param. Check the authorization docs: https://identityserver.github.io/Documentation/docsv2/endpoints/authorization.html

**Comment by farshidpourlatifi** [comment]
Hi Brock,

First of all, Thanks for the great job man.

> If this is not working for you, then you need sort this part out first.

It dose work like a charm, and it dose what it should, what I meant was it dose not fit into my scenario. not to mention that I understand that this part should be functional first to go on...

After I posted this, I implemented the script idea and that part works too.
But, what you mentioned about the authorization endpoint, together with prompt=none is something I didn't know, will switch my solution to use this end point.

Thanks for your time and efforts.
